envhelp/makeconfig

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+#
+# Standalone wrapper for generating a test configuration.
+
+set -e
+
+SCRIPT_PATH=$(realpath "$0")
+SCRIPT_BASE=$(dirname -- "$SCRIPT_PATH")
+MIG_BASE=$(realpath "$SCRIPT_BASE/..")
+PYTHON_BIN=${PYTHON_BIN:-"$SCRIPT_BASE/venv/bin/python3"}
```

Comment:
I wonder if this line should actually be delayed slightly until after next line and use the extracted `PY` value to switch between python2 and python3 rather than hard-coding `python3`? That would seem what one wants if requesting `PY=2` .

Reply:
Great that you asked about this because it looks like that ought to work but won't and doing this is very intentional. But this is subtle so worth explaining :)

So, the first hint of what's going on here is [here](https://github.com/ucphhpc/migrid-sync/pull/77/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR54). What you're seeing there is CI, when running in python 2 mode, invoking this script but directly setting the PYTHON_BIN to `python` aka 2.7. In CI, we need to build this configuration when we are _already_ inside a container, but `./envhelp/python2` is wired to execute whatever was requested within a container. We could patch up the envhelp wrapper but the result of my attempt to do so was a mess. Note also that locally we do generally need makeconfig to run under both versions - for the repo to be functional we have a hard requirement on 3 being available to bootstrap, so this works.

Reply:
This also prompted me to do another thing - it was already on my mind yesterday that `PY=2` was being used in two different ways in the presence of these changes and I'd considered making it an argument to makeconfig instead. Your question had me conclude this was in fact confusing, so have changed it:
https://github.com/ucphhpc/migrid-sync/pull/77/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR54

Reply:
Alright, I think it makes sense with that distinction.